Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -36,6 +36,20 @@ void Status::Update(const Status& new_status) {
   }
 }
 
+Status& Status::CausedBy(const Status& cause) {
+  if (cause.ok()) {
+    return *this;
+  }
+
+  if (ok()) {
+    *this = cause;
+    return *this;
+  }
+
+  absl::StrAppend(&state_->msg, ": ", cause.error_message());
```

Comment:
Consider adding a self test, i.e.

HARD_ASSERT(this != &cause, "A Status can't be CausedBy itself!");

Reason (besides the fact that it's silly) is that if absl::StrAppend is passed a reference to the string to be appended to, it yields undefined behaviour.

"WARNING: `StrAppend(&str, a, b, c, ...)` requires that none of the a, b, c, parameters be a reference into str."

https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_cat.h#L334

Also consider adding this to the docstring.